pkg/controller/autoimport/autoimport_controller.go

```diff
@@ -170,3 +193,31 @@ func (r *ReconcileAutoImport) Reconcile(ctx context.Context, request reconcile.R
 
 	return result, iErr
 }
+
+func (r *ReconcileAutoImport) getGenerateClientHolderFuncFromAutoImportSecret(secret *corev1.Secret) (helpers.GenerateClientHolderFunc, error) {
+	if _, hasKubeConfig := secret.Data["kubeconfig"]; hasKubeConfig {
+		return helpers.GenerateImportClientFromKubeConfigSecret, nil
+	}
+
+	_, hasKubeAPIToken := secret.Data["token"]
+	_, hasKubeAPIServer := secret.Data["server"]
+	if hasKubeAPIToken && hasKubeAPIServer {
+		return helpers.GenerateImportClientFromKubeTokenSecret, nil
+	}
+
+	_, hasOCMAPIToken := secret.Data["ocm_api_token"]
+	rosaClusterID, hasRosaClusterID := secret.Data["ocm_rosa_cluster_id"]
+	if hasOCMAPIToken && hasRosaClusterID {
+		getter, ok := r.rosaKubeConfigGetters[string(rosaClusterID)]
+		if !ok {
+			getter = helpers.NewRosaKubeConfigGetter()
+			r.rosaKubeConfigGetters[string(rosaClusterID)] = getter
+		}
+
+		return func(secret *corev1.Secret) (reconcile.Result, *helpers.ClientHolder, meta.RESTMapper, error) {
+			return helpers.GenerateImportClientFromRosaCluster(getter, secret)
+		}, nil
+	}
+
+	return nil, fmt.Errorf("kubeconfig or token/server pair or rosa cluster info is required")
```

Comment:
should return an import faiiled condition.

Reply:
If failed, we have a condition to represent this error, https://github.com/stolostron/managedcluster-import-controller/pull/302/files#diff-03d9e6ab77a9fa69840a6e5dfd9f090260a3c4707ce3d5c6f7aae0502663ea4cR142